web/src/main/java/org/springframework/security/web/server/authentication/DelegatingServerAuthenticationConverter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.authentication;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.util.Assert;
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * A {@link ServerAuthenticationConverter} that delegates to other
+ * {@link ServerAuthenticationConverter} instances.
+ *
+ * @author DingHao
+ * @since 6.3
+ */
+public class DelegatingServerAuthenticationConverter implements ServerAuthenticationConverter {
+
+	private final List<ServerAuthenticationConverter> delegates;
+
+	private boolean continueOnError = true;
+
+	private final Log logger = LogFactory.getLog(getClass());
+
+	public DelegatingServerAuthenticationConverter(ServerAuthenticationConverter... converters) {
+		this(List.of(converters));
+	}
+
+	public DelegatingServerAuthenticationConverter(List<ServerAuthenticationConverter> converters) {
+		Assert.notEmpty(converters, "converters cannot be null");
+		this.delegates = converters;
+	}
+
+	@Override
+	public Mono<Authentication> convert(ServerWebExchange exchange) {
```

Comment:
In DelegatingReactiveAuthenticationManager, the continueOnError flag added optional behavior. In the case of DelegatingServerAuthenticationConverter, I think that the initial behavior of the component should be concatMapDelayError, and the continueOnError flag is not needed in this case, that is:
```
	@Override
	public Mono<Authentication> convert(ServerWebExchange exchange) {
		return Flux.fromIterable(delegates)
				.concatMapDelayError(c -> c.convert(exchange)
						.doOnError(logger::debug)
				)
				.next();
	}
```
@jzheaux what do you think about it?

Reply:
I think that it should fail when there is an error by default. The reason is that Spring Security delegates should return `null` when they don't have anything to do and reserve the right to error if they deem the contents to somehow be insecure.